crates/starknet-devnet-core/src/starknet/get_class_impls.rs

```diff
@@ -1,38 +1,44 @@
+use blockifier::state::state_api::StateReader;
 use starknet_rs_core::types::BlockId;
 use starknet_types::contract_address::ContractAddress;
 use starknet_types::contract_class::ContractClass;
 use starknet_types::felt::{ClassHash, Felt};
 
-use crate::error::{DevnetResult, Error};
+use crate::error::{DevnetResult, Error, StateError};
 use crate::starknet::Starknet;
-use crate::traits::DevnetStateReader;
+use crate::state::CustomStateReader;
 
 pub fn get_class_hash_at_impl(
-    starknet: &Starknet,
+    starknet: &mut Starknet,
     block_id: &BlockId,
     contract_address: ContractAddress,
 ) -> DevnetResult<ClassHash> {
-    let state = starknet.get_state_at(block_id)?;
-    let class_hash = state.state.state.class_hash_at(&contract_address);
+    let state = starknet.get_mut_state_at(block_id)?;
+    state.assert_contract_deployed(contract_address)?;
```

Comment:
assert_contract_deployed could be omitted, because internally it uses is_contract_deployed which uses get_class_hash_at

Reply:
`assert_contract_deployed` uses self.get_class_hash_at and checks if its not StarkFelt::Zero. After the method invokation the same check is done again

Reply:
Ah, alright, you mean omitting the assertion just in this place. I thought you meant generally dropping the `assert_contract_deployed` logic. Will address this.

Reply:
Removed in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/329/commits/0083e8f90b1370896120c3aa5edfda535b256a77